packages/core/src/providers/studio/index.ts

```diff
@@ -120,74 +124,89 @@ export class StudioLivepeerProvider extends BaseLivepeerProvider {
+          { tusEndpoint: string; asset: { id: string } },
+          Omit<CreateAssetSource, 'file'>
+        >('/asset/request-upload', {
+          json: {
+            name: source.name,
+            meta: source.meta,
+          },
+          headers: this._defaultHeaders,
+        });
+
+        const {
+          tusEndpoint,
+          asset: { id: assetId },
+        } = uploadReq;
+
+        await new Promise<void>((resolve, reject) => {
+          const upload = new tus.Upload(source.file!, {
+            endpoint: tusEndpoint,
+            metadata: {
+              id: assetId,
+            },
+            ...(source.file instanceof File
+              ? null
+              : { chunkSize: 5 * 1024 * 1024 }),
+
+            onError: (error) => {
+              console.log('Failed because: ', error);
+            },
+            onProgress(bytesSent, bytesTotal) {
+              const progress = bytesSent / bytesTotal;
+
+              uploadProgress.sources[index] = {
+                name: source.name,
+                progress,
+              };
+
+              uploadProgress.average =
+                uploadProgress.sources.reduce(
+                  (acc, { progress }) => acc + progress,
+                  0,
+                ) / uploadProgress.sources.length;
+
+              onUploadProgress?.(uploadProgress);
+            },
+
+            onSuccess() {
+              resolve();
+            },
+          });
+
+          upload
+            .findPreviousUploads()
+            .then((previousUploads) => {
+              if (previousUploads?.length > 0 && previousUploads[0]) {
+                upload.resumeFromPreviousUpload(previousUploads[0]);
+              }
+              upload.start();
+            })
+            .catch(reject);
+        });
+
+        return this.getAsset(assetId);
+      }),
+    );
 
-      const upload = new tus.Upload(args.file, {
-        endpoint: tusEndpoint,
-        metadata: {
-          id: assetId,
-        },
-        uploadSize: args.uploadSize,
-        // Chunk size is required if input is a stream (and S3 min is 5MB), but
-        // not recommended if it is a file.
-        ...(args.file instanceof File ? null : { chunkSize: 5 * 1024 * 1024 }),
-
-        onError(error) {
-          reject(error);
-        },
-        onProgress(bytesSent, bytesTotal) {
-          args?.onUploadProgress?.(bytesSent / bytesTotal);
-        },
-        onSuccess() {
-          args?.onUploadProgress?.(1);
-          return resolve();
-        },
-      });
-
-      upload
-        .findPreviousUploads()
-        .then((previousUploads) => {
-          if (previousUploads?.length > 0 && previousUploads[0]) {
-            upload.resumeFromPreviousUpload(previousUploads[0]);
-          }
-          upload.start();
-        })
-        .catch(reject);
+    return assets.map((asset) => {
+      if (asset.status === 'fulfilled') {
+        return asset.value;
+      } else {
+        throw asset.reason;
```

Comment:
Great, yeah I like this functionality! We can expand further in the future to add a feature for "don't throw on error"